### PR TITLE
New Factory global flags merge function

### PR DESCRIPTION
**Pull request by helen-frank**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
These code comes multiple times, we may extract them as one function.
**Which issue(s) this PR fixes**:
Fixes #
Port #2611 
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/options/global.go [hunk 1]
```diff
@@ -27,3 +28,10 @@ func (o *GlobalCommandOptions) AddFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&o.KubeConfig, "kubeconfig", "", "Path to the control plane kubeconfig file.")
 	flags.StringVar(&o.KarmadaContext, "karmada-context", "", "Name of the cluster context in control plane kubeconfig file.")
 }
+
+// AddFlags adds flags to the specified FlagSet.
+func AddFlags(cf *genericclioptions.ConfigFlags, flags *pflag.FlagSet) {
+	flags.StringVar(cf.KubeConfig, "kubeconfig", *cf.KubeConfig, "Path to the kubeconfig file to use for CLI requests.")
+	flags.StringVar(cf.Context, "karmada-context", *cf.Context, "The name of the kubeconfig context to use")
+	flags.StringVarP(cf.Namespace, "namespace", "n", *cf.Namespace, "If present, the namespace scope for this CLI request")
```
Comment (lonelyCZ) [inline pkg/karmadactl/options/global.go:36]: The `namespace` is not general, like `cordon`, `join`, `token` command etc, so I suggest to only retain `kubeconfig` and `karmada-context`.

Comment (helen-frank) [reply pkg/karmadactl/options/global.go]: fixed this and `taint` @lonelyCZ

File: pkg/karmadactl/options/global.go [hunk 2]
```diff
@@ -27,3 +28,9 @@ func (o *GlobalCommandOptions) AddFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&o.KubeConfig, "kubeconfig", "", "Path to the control plane kubeconfig file.")
 	flags.StringVar(&o.KarmadaContext, "karmada-context", "", "Name of the cluster context in control plane kubeconfig file.")
 }
+
+// AddFlags adds flags to the specified FlagSet.
+func AddFlags(cf *genericclioptions.ConfigFlags, flags *pflag.FlagSet) {
+	flags.StringVar(cf.KubeConfig, "kubeconfig", *cf.KubeConfig, "Path to the kubeconfig file to use for CLI requests.")
+	flags.StringVar(cf.Context, "karmada-context", *cf.Context, "The name of the kubeconfig context to use")
+}
```
[the remark this answers is not in the document]
Comment (helen-frank) [reply pkg/karmadactl/options/global.go:36]: This is acceptable, `AddKubeConfigFlags` as a `defaultConfigFlags` specific function

File: pkg/karmadactl/options/global.go [hunk 3]
```diff
@@ -1,9 +1,39 @@
...
+)
 
 // DefaultKarmadaClusterNamespace defines the default namespace where the member cluster secrets are stored.
 const DefaultKarmadaClusterNamespace = "karmada-cluster"
 
 // DefaultKarmadactlCommandDuration defines the default timeout for karmadactl execute
 const DefaultKarmadactlCommandDuration = 60 * time.Second
+
+// GlobalCommandOptions holds the configuration shared by the all sub-commands of `karmadactl`.
+type GlobalCommandOptions struct {
```
Comment (lonelyCZ) [inline pkg/karmadactl/options/global.go:17]: We don't need `GlobalCommandOptions` anywhere.

Comment (helen-frank) [reply pkg/karmadactl/options/global.go]: fixed

**Comment by lonelyCZ** [comment]
I just tested it, and it worked fine in my env.

Hi, @RainbowMango , please help to retrigger the workflows.

**Comment by lonelyCZ** [comment]
PTAL

/cc @carlory

File: pkg/karmadactl/promote.go [hunk 4]
```diff
@@ -85,8 +86,8 @@ func NewCmdPromote(f util.Factory, parentCommand string) *cobra.Command {
 	flag := cmd.Flags()
 	opts.AddFlags(flag)
 
-	flag.StringVar(defaultConfigFlags.KubeConfig, "kubeconfig", *defaultConfigFlags.KubeConfig, "Path to the kubeconfig file to use for CLI requests.")
-	flag.StringVar(defaultConfigFlags.Context, "karmada-context", *defaultConfigFlags.Context, "The name of the kubeconfig context to use")
+	flag.StringVar(options.DefaultConfigFlags.KubeConfig, "kubeconfig", *options.DefaultConfigFlags.KubeConfig, "Path to the kubeconfig file to use for CLI requests.")
```
Comment (lonelyCZ) [inline pkg/karmadactl/promote.go:89]: Why not `options.AddKubeConfigFlags()`?

Comment (helen-frank) [reply pkg/karmadactl/promote.go]: Sorry for the omission, now it has been replaced.

**Comment by lonelyCZ** [comment]
Please take a look~

/cc @carlory

**Comment by lonelyCZ** [comment]
It works fine in my env.

/lgtm
/approve

**Comment by XiShanYongYe-Chang** [comment]
Change the label to the correct one.
/kind cleanup
/remove-kind feature

**Pull request merged**